api/src/com/cloud/agent/api/to/VirtualMachineTO.java

```diff
@@ -69,6 +69,7 @@
     String configDriveIsoRootFolder = null;
     String configDriveIsoFile = null;
 
+    Map<String, String> guestOsDetails = null;
```

Comment:
To avoid potential NPEs, construct a hashmap with no items, such as
`Map<String, String> guestOsDetails = new Hashmap();`